index.js

```diff
@@ -209,6 +209,47 @@ function filePathFactory(file) {
 }
 
 /**
+ * If the vFile has a extension, it will be prefixed with the filename,
+ * if applicable. If the vFile does not have extension,
+ * only `filename` will be returned.
+ * Otherwise, an empty string is returned.
+ *
+ * @private
+ * @param {VFile} file - Virtual file.
+ * @return {Function} - `fileName` getter.
+ */
+function fileBasenameFactory(file) {
+    /**
+     * Get the filename with extantion.
+     *
+     * @example
+     *   var file = new VFile({
+     *     'filename': 'example',
+     *     'extension': 'txt'
+     *   });
+     *
+     *   String(file.fileName); // example.txt
+     *   file.fileName() // example.txt
+     *
```

Comment:
Could you update the example here? Preferrable, it would include a `directory` too, to show the difference between both.